src/3D/writeCGNS.F90

```diff
@@ -92,6 +92,9 @@ subroutine writeCGNS(fileName)
         ! Copy values and write:
         do k=1,N
 
+           ! Synchronize before scattering
+           call mpi_barrier(hyp_comm_world, ierr)
```

Comment:
What is the reasoning behind this barrier?

Reply:
The tests were failing on the TACC image without this barrier. I think it is related to the newer Intel MPI version used on that image. We did not see this problem before because all the tests ran on 1 proc.

Does it make sense to call barrier before a PETSc generalized scatter? I'm not convinced this is a bug in pyHyp as opposed to something wrong with the TACC image or Intel MPI.

Reply:
Would have expected it to be fine even w/o the barrier but I dont know enough. If it fixes an issue on TACC, then sure.

Reply:
Yeah this is a bit weird, Sabet and I had some conversations about this but I don't really understand what's going on. Maybe it's an MPI implementation-specific behaviour?

Reply:
So I can try to read and understand the scatter stuff and its usage, but it will take a few more days probably. To me this is not too performance critical since these meshes are run once and not in the loop etc.

Reply:
Yeah I thought scatter is blocking already, but according to @sseraj he tried many combinations and this was the only one that worked consistently.